sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2377,14 +2375,14 @@ class Analyzer(override val catalogManager: CatalogManager)
     def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsUpWithPruning(
       _.containsPattern(PLAN_EXPRESSION), ruleId) {
       case j: LateralJoin if j.left.resolved =>
-        resolveSubQueries(j, j.children)
+        resolveSubQueries(j)
```

Comment:
LateralJoin node has `resolveChildren` overridden to use both `left.output` and `right.plan.output` so that it can be used to resolve the join conditions for LateralJoin. Maybe we can modify this method to condition on whether right tree is resolved or not.
https://github.com/apache/spark/blob/6ca56b01dc53259a57d92a8755660bcfb3a12612/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala#L1455-L1469